cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -2199,6 +2211,12 @@
          (MInst.CSNeg dst cond if_true if_false)
          dst)))
 
+;; Helper for generating `MInst.CCmp` instructions.
+(decl ccmp (OperandSize Reg Reg NZCV Cond) ConsumesFlags)
```

Comment:
This definitely *works* but it seems like we might be able to be slightly more descriptive and precise with the types here: `ccmp` both consumes and produces flags. Maybe we could:

- Define a new `ProducesFlags` enum arm that holds two instructions in sequence;
- Define a `ccmp` helper that takes an existing `ProducesFlags` and returns a `ProducesFlags`, gluing the `ccmp` onto the end of the existing producer as an "update" to the flags. (It's OK if we only handle the two-instruction case here for now; we can extend the enum later if necessary)

Thoughts?